### PR TITLE
MGMT-10830 - Create base class for test-infra containerized controllers

### PR DESCRIPTION
**Pull request by eliorerz**
Join base logic for test-infra containerized controllers
After adding tang support to test-infra we have 3 duplicated code/behaviors on the containerized controller: ipxe_controller, tang_controller, and on proxy_controller. This PR move the base controller behavior under new base class.

/cc @osherdp

### CODE REVIEW

**Comment by eliorerz** [comment]
/hold
Wait for `e2e-metal-assisted-ipxe`, `e2e-metal-assisted-tang` and  `e2e-metal-assisted-ipv6` to pass

File: src/assisted_test_infra/test_infra/controllers/containerized_controller.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+from assisted_test_infra.test_infra.utils import utils
+from service_client import log
+
+
+class ContainerizedController(ABC):
+    def __init__(self, name: str, port: int, image: str) -> None:
+        self._name = name
+        self._port = port
+        self._image = image
+        self._extra_flags = []
```
Comment (osherdp) [inline src/assisted_test_infra/test_infra/controllers/containerized_controller.py:12]: maybe enable setting ``_extra_flags`` from ctor parameters as well

File: src/assisted_test_infra/test_infra/controllers/proxy_controller/proxy_controller.py [hunk 2]
```diff
@@ -6,82 +6,71 @@
...
             self.authenticated = authenticated
-            self.image = "quay.io/sameersbn/squid"
-            self.dir = dir
+            self.image = ""
             self._is_ipv6 = is_ipv6
             self._set_server_address(host_ip)
             self._create_conf_from_template(denied_port=denied_port)
             self._create_user_file_for_auth()
-            self._run_proxy_server()
+            self.run()
```
Comment (osherdp) [inline src/assisted_test_infra/test_infra/controllers/proxy_controller/proxy_controller.py:40]: kind of weird to do the ``run`` from the c'tor. can we move it from this scope?

**Review by osherdp — CHANGES_REQUESTED** [review]

File: src/assisted_test_infra/test_infra/controllers/proxy_controller/proxy_controller.py (hunk 2 above)
Comment (eliorerz) [reply src/assisted_test_infra/test_infra/controllers/proxy_controller/proxy_controller.py:40]: I didn't want to change its behavior , you think it's ok to do it inside this PR?

Comment (osherdp) [reply src/assisted_test_infra/test_infra/controllers/proxy_controller/proxy_controller.py:40]: I think so. it's just moving one line

**Review by osherdp — APPROVED** [review]

**Pull request merged**